phases/19-capstone-projects/49-lm-eval-harness/code/main.py

```
@@ -0,0 +1,493 @@
+    Targets is a list of stringified expected results; extras carries a
+    list of (input, output) pairs the function is checked against. The
+    code runs in a stripped builtins namespace so it cannot reach the
+    filesystem or network.
+    """
+    pairs = extras.get("io_pairs") or []
+    if not isinstance(pairs, list) or not pairs:
+        return 0.0
+    safe_globals = {"__builtins__": {"range": range, "len": len, "min": min, "max": max, "abs": abs, "int": int, "float": float}}
+    local: Dict[str, object] = {}
+    try:
+        exec(prediction, safe_globals, local)
```

Ruff (0.15.14)

[error] Use of `exec` detected

(S102)